### DISCUSSION TITLE
Huge lags reading from usb on raspberry 3

### DISCUSSION THREAD

**Discussion by SleepWalker** [Koenkk/zigbee2mqtt#13227] [Devices]
I have zigbee coordinator based on CC2652P. When I connecting it to my laptop it works just fine. But when connected to raspberry pi it takes ages to get response from e.g. MI button click.

I've found out that when I starting network map from zigbee2mqtt UI all the pending clicks are immediately received and forwarded to mqtt. I've played around with code and found that it's enough to periodically request `lqi` from coordinator device in order to "fix" an issue, e.g.:

```js
setInterval(
  () => this.zigbee.devices().filter((d) => d.zh.type === 'Coordinator').forEach(i => i.zh.lqi()),
  1000
);
```

But it doesn't look like a good fix. Especially in docker environment.

I've tried to disable usb suspending, tried to use extension usb cable and powered usb hub. But without any luck. Is there any setting that can be tweaked so that Coordinator will become responsive?

---

UPDATE: another one observation. When I testing coordinator with Mi Button, I've noticed that it stucks for 7 clicks and after 8th I'm getting back all the 8 clicks at a time.

**Comment by Koenkk** [comment]
Sounds like https://github.com/Koenkk/zigbee2mqtt/issues/8663

**Reply by SleepWalker** [reply]
Wow. Amazing 🔥This have solved the issue. I've spent two days trying to make it work 😅 I'm so happy. Thank you very much
